### PR TITLE
Support KiND cluster for local dev environment #800

### LINKED ISSUE #800: Support KiND cluster for local dev environment
## Describe the proposal

This is the proposal to support [KiND](https://kind.sigs.k8s.io/). Unlike Minikube, KiND is running all k8s controllers in the single docker container so that it require to create VM.

- [ ] Provide the document to use KiND for local dev environment
- [ ] Update sample docs to support KiND

### PR DESCRIPTION
**Pull request by robsondepaula**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

This is a WIP PR to address the lack of setup instructions for kubernetes clusters using KiND.

## Issue reference

 #800

### CODE REVIEW

**Comment by yaron2** [comment]
@robsondepaula see the failed links validation check - you should probably mark localhost:9999 as markdown `localhost:9999`.

**Comment by robsondepaula** [comment]
@yaron2 it is fixed, thanks for the feedback.

**Comment by msfussell** [comment]
@robsondepaula - Taking a look at this and just going through the steps to validate

File: daprdocs/content/en/operations/hosting/kubernetes/cluster/setup-kind.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+---
+type: docs
+title: "Setup a KiND cluster"
```
Comment (greenie-msft) [inline daprdocs/content/en/operations/hosting/kubernetes/cluster/setup-kind.md]:
```suggestion
title: "Set up a KiND cluster"
```

File: daprdocs/content/en/operations/hosting/kubernetes/cluster/setup-kind.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+linkTitle: "KiND"
+weight: 1100
+description: >
+  How to setup Dapr on a KiND cluster.
```
Comment (greenie-msft) [inline daprdocs/content/en/operations/hosting/kubernetes/cluster/setup-kind.md]:
```suggestion
  How to set up Dapr on a KiND cluster.
```

**Comment by greenie-msft** [comment]
Tested e2e. Looks good to me.

**Review by greenie-msft — APPROVED** [review]

**Review by msfussell — APPROVED** [review]
LGTM

**Pull request merged**